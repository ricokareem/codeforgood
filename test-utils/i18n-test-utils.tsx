import type React from "react"
import { I18nextProvider } from "react-i18next"
import i18n from "i18next"
import { initReactI18next } from "react-i18next"

// Create a test instance of i18next
const createTestI18n = (language: string, resources: any) => {
  const i18nInstance = i18n.createInstance()
  i18nInstance.use(initReactI18next).init({
    lng: language,
    fallbackLng: "en",
    resources,
    interpolation: {
      escapeValue: false,
    },
  })
  return i18nInstance
}

// Wrapper component for testing with i18n
export const TestI18nProvider = ({
  children,
  language = "en",
  resources,
}: {
  children: React.ReactNode
  language?: string
  resources: any
}) => {
  const i18nInstance = createTestI18n(language, resources)
  return <I18nextProvider i18n={i18nInstance}>{children}</I18nextProvider>
}
