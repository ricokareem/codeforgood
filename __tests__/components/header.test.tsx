import { render } from "@testing-library/react"
import Header from "@/components/header"
import { TestI18nProvider } from "../../test-utils/i18n-test-utils"

// Mock Cookies
jest.mock("js-cookie", () => ({
  get: jest.fn(),
  set: jest.fn(),
}))

// Mock translations
const translations = {
  en: {
    translation: {
      menu: {
        about: "About",
        experience: "Experience",
        skills: "Skills",
        projects: "Projects",
        education: "Education",
        selectLanguage: "Select language",
        community: "Community",
      },
    },
  },
  de: {
    translation: {
      menu: {
        about: "Über mich",
        experience: "Erfahrung",
        skills: "Fähigkeiten",
        projects: "Projekte",
        education: "Ausbildung",
        selectLanguage: "Sprache auswählen",
        community: "Gemeinschaft",
      },
    },
  },
  es: {
    translation: {
      menu: {
        about: "Acerca de",
        experience: "Experiencia",
        skills: "Habilidades",
        projects: "Proyectos",
        education: "Educación",
        community: "Comunidad",
      },
    },
  },
}

describe("Header Component", () => {
  it("renders correctly in English", () => {
    const { container } = render(
      <TestI18nProvider language="en" resources={translations}>
        <Header />
      </TestI18nProvider>,
    )
    expect(container).toMatchSnapshot()
  })

  it("renders correctly in German", () => {
    const { container } = render(
      <TestI18nProvider language="de" resources={translations}>
        <Header />
      </TestI18nProvider>,
    )
    expect(container).toMatchSnapshot()
  })

  it("renders correctly in Spanish", () => {
    const { container } = render(
      <TestI18nProvider language="es" resources={translations}>
        <Header />
      </TestI18nProvider>,
    )
    expect(container).toMatchSnapshot()
  })
})
