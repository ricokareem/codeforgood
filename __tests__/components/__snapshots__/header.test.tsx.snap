// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders correctly in English 1`] = `
<div>
  <header
    class="fixed w-full bg-background/80 backdrop-blur-sm z-50 border-b border-border"
  >
    <nav
      class="container mx-auto px-4 py-4 flex justify-between items-center"
    >
      <a
        class="text-2xl font-bold text-foreground"
        href="/"
      >
        CFG
      </a>
      <div
        class="flex items-center space-x-4"
      >
        <div
          class="relative"
        >
          <select
            aria-label="Select language"
            class="appearance-none bg-accent text-accent-foreground py-2 pl-3 pr-10 rounded-md focus:outline-none focus:ring-2 focus:ring-primary"
            style="width: 140px; height: 40px;"
          >
            <option
              value="en"
            >
              ENGLISH ðŸ‡¬ðŸ‡§
            </option>
            <option
              value="de"
            >
              DEUTSCH ðŸ‡©ðŸ‡ª
            </option>
            <option
              value="es"
            >
              ESPAÃ‘OL ðŸ‡ªðŸ‡¸
            </option>
          </select>
          <div
            class="pointer-events-none absolute inset-y-0 right-0 flex items-center px-2 text-foreground"
          >
            <svg
              class="fill-current h-4 w-4"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z"
              />
            </svg>
          </div>
        </div>
        <button
          class="md:hidden text-foreground"
        >
          <svg
            class="w-6 h-6"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
          >
            <path
              d="M4 6h16M4 12h16M4 18h16"
            />
          </svg>
        </button>
        <ul
          class="md:flex space-x-4 hidden"
        >
          <li>
            <a
              class="text-muted-foreground hover:text-foreground transition-colors"
              href="#about"
            >
              About
            </a>
          </li>
          <li>
            <a
              class="text-muted-foreground hover:text-foreground transition-colors"
              href="#experience"
            >
              Experience
            </a>
          </li>
          <li>
            <a
              class="text-muted-foreground hover:text-foreground transition-colors"
              href="#skills"
            >
              Skills
            </a>
          </li>
          <li>
            <a
              class="text-muted-foreground hover:text-foreground transition-colors"
              href="#projects"
            >
              Projects
            </a>
          </li>
          <li>
            <a
              class="text-muted-foreground hover:text-foreground transition-colors"
              href="#education"
            >
              Education
            </a>
          </li>
          <li>
            <a
              class="text-muted-foreground hover:text-foreground transition-colors"
              href="#community"
            >
              Community
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </header>
</div>
`;

exports[`Header Component renders correctly in German 1`] = `
<div>
  <header
    class="fixed w-full bg-background/80 backdrop-blur-sm z-50 border-b border-border"
  >
    <nav
      class="container mx-auto px-4 py-4 flex justify-between items-center"
    >
      <a
        class="text-2xl font-bold text-foreground"
        href="/"
      >
        CFG
      </a>
      <div
        class="flex items-center space-x-4"
      >
        <div
          class="relative"
        >
          <select
            aria-label="Sprache auswÃ¤hlen"
            class="appearance-none bg-accent text-accent-foreground py-2 pl-3 pr-10 rounded-md focus:outline-none focus:ring-2 focus:ring-primary"
            style="width: 140px; height: 40px;"
          >
            <option
              value="en"
            >
              ENGLISH ðŸ‡¬ðŸ‡§
            </option>
            <option
              value="de"
            >
              DEUTSCH ðŸ‡©ðŸ‡ª
            </option>
            <option
              value="es"
            >
              ESPAÃ‘OL ðŸ‡ªðŸ‡¸
            </option>
          </select>
          <div
            class="pointer-events-none absolute inset-y-0 right-0 flex items-center px-2 text-foreground"
          >
            <svg
              class="fill-current h-4 w-4"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z"
              />
            </svg>
          </div>
        </div>
        <button
          class="md:hidden text-foreground"
        >
          <svg
            class="w-6 h-6"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
          >
            <path
              d="M4 6h16M4 12h16M4 18h16"
            />
          </svg>
        </button>
        <ul
          class="md:flex space-x-4 hidden"
        >
          <li>
            <a
              class="text-muted-foreground hover:text-foreground transition-colors"
              href="#about"
            >
              Ãœber mich
            </a>
          </li>
          <li>
            <a
              class="text-muted-foreground hover:text-foreground transition-colors"
              href="#experience"
            >
              Erfahrung
            </a>
          </li>
          <li>
            <a
              class="text-muted-foreground hover:text-foreground transition-colors"
              href="#skills"
            >
              FÃ¤higkeiten
            </a>
          </li>
          <li>
            <a
              class="text-muted-foreground hover:text-foreground transition-colors"
              href="#projects"
            >
              Projekte
            </a>
          </li>
          <li>
            <a
              class="text-muted-foreground hover:text-foreground transition-colors"
              href="#education"
            >
              Ausbildung
            </a>
          </li>
          <li>
            <a
              class="text-muted-foreground hover:text-foreground transition-colors"
              href="#community"
            >
              Gemeinschaft
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </header>
</div>
`;

exports[`Header Component renders correctly in Spanish 1`] = `
<div>
  <header
    class="fixed w-full bg-background/80 backdrop-blur-sm z-50 border-b border-border"
  >
    <nav
      class="container mx-auto px-4 py-4 flex justify-between items-center"
    >
      <a
        class="text-2xl font-bold text-foreground"
        href="/"
      >
        CFG
      </a>
      <div
        class="flex items-center space-x-4"
      >
        <div
          class="relative"
        >
          <select
            aria-label="Select language"
            class="appearance-none bg-accent text-accent-foreground py-2 pl-3 pr-10 rounded-md focus:outline-none focus:ring-2 focus:ring-primary"
            style="width: 140px; height: 40px;"
          >
            <option
              value="en"
            >
              ENGLISH ðŸ‡¬ðŸ‡§
            </option>
            <option
              value="de"
            >
              DEUTSCH ðŸ‡©ðŸ‡ª
            </option>
            <option
              value="es"
            >
              ESPAÃ‘OL ðŸ‡ªðŸ‡¸
            </option>
          </select>
          <div
            class="pointer-events-none absolute inset-y-0 right-0 flex items-center px-2 text-foreground"
          >
            <svg
              class="fill-current h-4 w-4"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z"
              />
            </svg>
          </div>
        </div>
        <button
          class="md:hidden text-foreground"
        >
          <svg
            class="w-6 h-6"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
          >
            <path
              d="M4 6h16M4 12h16M4 18h16"
            />
          </svg>
        </button>
        <ul
          class="md:flex space-x-4 hidden"
        >
          <li>
            <a
              class="text-muted-foreground hover:text-foreground transition-colors"
              href="#about"
            >
              Acerca de
            </a>
          </li>
          <li>
            <a
              class="text-muted-foreground hover:text-foreground transition-colors"
              href="#experience"
            >
              Experiencia
            </a>
          </li>
          <li>
            <a
              class="text-muted-foreground hover:text-foreground transition-colors"
              href="#skills"
            >
              Habilidades
            </a>
          </li>
          <li>
            <a
              class="text-muted-foreground hover:text-foreground transition-colors"
              href="#projects"
            >
              Proyectos
            </a>
          </li>
          <li>
            <a
              class="text-muted-foreground hover:text-foreground transition-colors"
              href="#education"
            >
              EducaciÃ³n
            </a>
          </li>
          <li>
            <a
              class="text-muted-foreground hover:text-foreground transition-colors"
              href="#community"
            >
              Comunidad
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </header>
</div>
`;
